tests/test_resolutions.py

```diff
@@ -221,7 +228,7 @@ def test_description(self) -> None:
 
         ctk = ContainsTheKey(test_key)
 
-        expected_description = f'Containing the key "{test_key}".'
+        expected_description = f'Containing the key {test_key!r}.'
```

Comment:
Seeing all of these changes is making me realize that i dun goofed when i wrote these tests this way. I thought i'd beaten out all the developer sensibilities when it comes to testing, but some still stuck with me i guess.

We should be writing the exact expected strings here so we don't have to make identical code changes to our tests when we make them in our code-under-test. We're just doing the same thing in two places and asking if the result is the same, obviously that's not a good test! 😩 

I'll make a different PR to update these tests—it's my silly mistake, i'll fix it.

Reply:
Agreed... however.
I think there is some wiggle room when it comes to using the language efficiently, no?
We aren't setting out to prove python does the right thing when you use `repr`.  We just expect that `repr` works.  We _do_ set out to prove that the string is formatted in a particular fashion.  We want to test _our_ code does the right thing.

`<soapbox>`
I don't see a problem with either of the following to construct an expected outcome.  Especially if it makes maintaining tests easier.
```python
expected = "Hello World!"
# vs
part1 = "Hello"
part2 = " World!"
expected = part1 + part2
```
I also don't a huge leap in mental logic between the above and this:
```python
expected = "HELLO WORLD!"
# vs
part1 = "Hello"
part2 = " World!"
expected = part1.upper() + part2.upper()
```
Using string formatting in the test didn't feel like a violation of testing principles. 
`<\soapbox>`
With that said, I don't think it's a bad idea to have our expected strings be explicit.